.github/workflows/veracode-scans.yml

```diff
@@ -0,0 +1,126 @@
+          target_branch_name: "${{ inputs.target_branch_name }}"
+          cache_key: '${{ inputs.cache_key }}'
+
+      - name: Set routing mode
+        id: routing-mode
+        if: inputs.event_name != 'pull_request' && (inputs.target_branch_name == 'develop' || inputs.target_branch_name == 'dev-[2-9][0-9].[0-9][0-9].x' || inputs.target_branch_name == 'master' || inputs.target_branch_name == '[2-9][0-9].[0-9][0-9].x')
+        run: echo "scan_type=maintenance" >> $GITHUB_OUTPUT
+
+  pipeline-scan:
+    runs-on: [self-hosted, veracode]
+    needs: [build]
+
+    steps:
+      - name: Get build binary
+        uses: actions/cache/restore@v3
+        with:
+          path: "${{ inputs.module_name }}-${{ inputs.target_branch_name }}.zip"
+          key: '${{ inputs.cache_key }}'
+
+      - name: Get current time
+        run: echo "$(date +'%Y-%m-%d %H:%M:%S')"
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 8
+
+      - name: Setup awscli #TODO: To install in veracode runner directly
+        run: |
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          sudo unzip -q awscliv2.zip
+          sudo ./aws/install
+
+      - name: Get baseline file
+        id: baseline_search
+        run: |
+          TARGET_S3="s3://centreon-veracode-reports/${{ inputs.module_name }}/${{ inputs.target_branch_name }}/results.json"
+          BASELINE_FILE="/tmp/results.json"
+          aws s3 ls "$TARGET_S3" || RESULT=$( echo $? )
+          if [[ $RESULT -eq 0 ]]; then
+            aws s3 cp "$TARGET_S3" "$BASELINE_FILE"
+            echo "baseline_file=$BASELINE_FILE" >> $GITHUB_ENV
+          else
+            echo "[DEBUG] - File not found in the bucket"
+            echo "baseline_file=" >> $GITHUB_ENV
+          fi
+
+      - name: Pipeline scan
+        uses: veracode/Veracode-pipeline-scan-action@v1.0.3
+        with:
+          vid: '${{ secrets.veracode_api_id }}'
+          vkey: '${{ secrets.veracode_api_key }}'
+          file: "${{ inputs.module_name }}-${{ inputs.target_branch_name }}.zip"
+          fail_on_severity: "${{ inputs.threshold_pipeline }}"
+          #fail_build: true #TODO: Do not fail the jobs until grace period ends
+          timeout: 60
+          baseline_file: "${{ env.baseline_file }}"
+
+      - name: Save baseline file
+        # only baseline file generated from a maintenance branch is saved
+        if: needs.build.outputs.scan_type == 'maintenance'
+        run: |
+          mv results.json /tmp
+          TARGET_S3="s3://centreon-veracode-reports/${{ inputs.module_name }}/${{ inputs.target_branch_name }}/results.json"
+          BASELINE_FILE="/tmp/results.json"
+          aws s3 cp "$BASELINE_FILE" "$TARGET_S3"
+
+      - name: Clean
+        run: rm -f results.json filtered_results.json /tmp/results.json
+
+      - name: Get current time
+        run: echo "$(date +'%Y-%m-%d %H:%M:%S')"
```

Comment:
```suggestion
```

Reply:
Get current time will be removed in the next ticket (sandbox scan). I will keep it for now